MapboxCoreNavigation/FreeDriveLocationManager.swift

```diff
@@ -0,0 +1,196 @@
+        let resolvedNavigator: Navigator
+        if let navigator = navigator {
+            resolvedNavigator = navigator
+        } else {
+            resolvedNavigator = createNavigator()
+        }
+        proxyDelegate = ProxyDelegate(navigator: resolvedNavigator)
+
+        super.init()
+
+        super.delegate = proxyDelegate
+    }
+
+    func setCustomLocation(_ location: CLLocation?) {
+        guard let location = location else { return }
+        stopUpdatingLocation()
+        stopUpdatingHeading()
+        proxyDelegate.locationManager(self, didUpdateLocations: [location])
+    }
+
+    // MARK: MGLLocationManager
+
+    override public var delegate: CLLocationManagerDelegate? {
+        get {
+            proxyDelegate.delegate
+        }
+        set {
+            proxyDelegate.delegate = newValue
+        }
+    }
+
+    public func debugView(onUpdated: ((CLLocationCoordinate2D, CLLocationCoordinate2D)->Void)?) -> UIView {
+        let debugView = FreeDriveDebugInfoView()
+        proxyDelegate.debugDelegate = debugView
+        debugView.onUpdated = onUpdated
+        return debugView
+    }
+
+    private class ProxyDelegate: NSObject, CLLocationManagerDelegate {
+        private var navNative: Navigator!
+        var delegate: CLLocationManagerDelegate?
+        weak var debugDelegate: FreeDriveDebugInfoListener?
+
+        required init(navigator: Navigator? = nil) {
+            self.navNative = navigator ?? Navigator()
+            super.init()
+        }
+
+        deinit {
+            self.navNative = nil
+        }
+
+        // MARK: CLLocationManagerDelegate
+
+        public func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
+            if let loc = locations.first {
+                navNative?.updateLocation(for: FixLocation(loc))
+                let projectedDate = Date()
+                let status = navNative.getStatusForTimestamp(projectedDate)
+                delegate?.locationManager?(manager, didUpdateLocations: [CLLocation(status.location)])
+
+                debugDelegate?.didGet(location: CLLocation(status.location), with: status.map_matcher_output.matches, for: loc)
+            }
+        }
+
+        public func locationManager(_ manager: CLLocationManager, didUpdateHeading newHeading: CLHeading) {
+            delegate?.locationManager?(manager, didUpdateHeading: newHeading)
+        }
+
+        public func locationManagerShouldDisplayHeadingCalibration(_ manager: CLLocationManager) -> Bool {
+            return delegate?.locationManagerShouldDisplayHeadingCalibration?(manager) ?? false
+        }
+
+        public func locationManager(_ manager: CLLocationManager, didFailWithError error: Error) {
+            delegate?.locationManager?(manager, didFailWithError: error)
+        }
+    }
+}
+
+private class Accounts: SkuTokenSource {
+    init() {
+        MBXAccounts.activateSKUID(.navigationUser)
```

Comment:
This should be unnecessary due to code in `+[MBXAccounts(CoreNavigationAdditions) load]` that automatically runs as soon as MapboxCoreNavigation.framework loads:

https://github.com/mapbox/mapbox-navigation-ios/blob/0fb35932f62387b96bac1f0653cf8766c3f6d3bb/MapboxCoreNavigation/MBXAccounts+CoreNavigationAdditions.m#L11

Reply:
The class will be removed, the idea was to use the SkuTokenProvider from #2533